### ISSUE TITLE
AttributeError: You cannot add attributes to <pyrfr.regression.forest_opts

### ISSUE THREAD

**Issue by CoderCoderCoder** [automl/auto-sklearn#305]
The following simple code:
```
import pandas
import numpy as np
import autosklearn.classification

url = "http://archive.ics.uci.edu/ml/machine-learning-databases/pima-indians-diabetes/pima-indians-diabetes.data"
names = ['preg', 'plas', 'pres', 'skin', 'test', 'mass', 'pedi', 'age', 'class']
dataframe = pandas.read_csv(url, names=names)
array = dataframe.values
X_train = array[:300,0:8]
Y_train = array[:300,8]
X_test = array[300:,0:8]
Y_test = array[300:,8]
print(X.shape)

X_train = np.ascontiguousarray(X_train)
cls = autosklearn.classification.AutoSklearnClassifier()
res = cls.fit(X_train, Y_train)
```

Results in:
AttributeError: You cannot add attributes to <pyrfr.regression.forest_opts; proxy of <Swig Object of type 'rfr::forests::forest_options< num_t,response_t,index_t > *' at 0x108eae870> >

How this can be solved?....

**Comment by CoderCoderCoder** [comment]
P.S. after changing to
```
X = array[:300,0:8]
Y = array[:300,8]
```
The error became 
BufferError: memoryview: underlying buffer is not C-contiguous

**Comment by CoderCoderCoder** [comment]
Full log:

AttributeError                            Traceback (most recent call last)
<ipython-input-29-f05fa11997e4> in <module>()
     16 X_train = np.ascontiguousarray(X_train)
     17 cls = autosklearn.classification.AutoSklearnClassifier()
---> 18 res = cls.fit(X_train, Y_train)
     19 #res = cls.show_models()
     20 print(res)

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/estimators.py in fit(self, X, y, metric, feat_type, dataset_name)
    403         return super(AutoSklearnClassifier, self).fit(X=X, y=y, metric=metric,
    404                                                       feat_type=feat_type,
--> 405                                                       dataset_name=dataset_name)
    406 
    407     def predict(self, X, batch_size=None, n_jobs=1):

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/estimators.py in fit(self, *args, **kwargs)
    302     def fit(self, *args, **kwargs):
    303         self._automl = self.build_automl()
--> 304         super(AutoSklearnEstimator, self).fit(*args, **kwargs)
    305 
    306     def fit_ensemble(self, y, task=None, metric=None, precision='32',

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/estimators.py in fit(self, *args, **kwargs)
     21 
     22     def fit(self, *args, **kwargs):
---> 23         self._automl.fit(*args, **kwargs)
     24 
     25     def refit(self, X, y):

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/estimators.py in fit(self, X, y, metric, loss, feat_type, dataset_name)
    544         y = self._process_target_classes(y)
    545 
--> 546         return self._automl.fit(X, y, task, metric, feat_type, dataset_name)
    547 
    548     def fit_ensemble(self, y, task=None, metric=None, precision='32',

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/automl.py in fit(self, X, y, task, metric, feat_type, dataset_name)
    193                                             dataset_name=dataset_name)
    194 
--> 195         return self._fit(loaded_data_manager, metric)
    196 
    197     def fit_automl_dataset(self, dataset, metric):

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/automl.py in _fit(self, datamanager, metric)
    447                                     configuration_mode=self._configuration_mode)
    448             self.runhistory_, self.trajectory_, self.fANOVA_input_ = \
--> 449                 _proc_smac.run_smbo()
    450             trajectory_filename = os.path.join(
    451                 self._backend.get_smac_output_directory(self._seed) + '_run1',

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/autosklearn/smbo.py in run_smbo(self)
    585             model = RandomForestWithInstances(types=types, bounds=bounds,
    586                                               #instance_features=meta_features_list,
--> 587                                               seed=1, num_trees=10)
    588             rh2EPM = RunHistory2EPM4Cost(num_params=num_params,
    589                                          scenario=self.scenario,

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/smac-0.5.0-py3.6.egg/smac/epm/rf_with_instances.py in __init__(self, types, bounds, num_trees, do_bootstrapping, n_points_per_tree, ratio_features, min_samples_split, min_samples_leaf, max_depth, eps_purity, max_num_nodes, seed, **kwargs)
     75         self.rf_opts = regression.forest_opts()
     76         self.rf_opts.num_trees = num_trees
---> 77         self.rf_opts.seed = seed
     78         self.rf_opts.do_bootstrapping = do_bootstrapping
     79         max_features = 0 if ratio_features >= 1.0 else \

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyrfr/regression.py in set_attr(self, name, value)
     93             set(self, name, value)
     94         else:
---> 95             raise AttributeError("You cannot add attributes to %s" % self)
     96     return set_attr
     97 

AttributeError: You cannot add attributes to <pyrfr.regression.forest_opts; proxy of <Swig Object of type 'rfr::forests::forest_options< num_t,response_t,index_t > *' at 0x10d8e9b40> >

**Comment by mfeurer** [comment]
Could you please provide an overview of your system (python version, operating system etc.)?

**Comment by CoderCoderCoder** [comment]
Hi Matthias,
thanks for your response. I managed to solve the issue only by the following steps:
- uninstalled swig, gcc installed by brew
- installed Anaconda (latest version)
- installed through anaconda swig and gcc
- uninstalled root-installed Python3.6.1 including all packages (very important! until I did it I couldn't get it to work for some weird reason)
- pip install pyrfr -> no errors this time
- pip install auto-sklearn -> no errors
- run the test code - no errors, code is executing

So I can only recommend everyone who is using Mac to work through Anaconda, it seems to solve a lot of nasty compatibility issues.

P.S. I am using MacOS  10.12.4.
